homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,41 @@
+#include "compression.hpp"
+
+compressedGrayscaleImage compressGrayscale(const grayscaleImage& bitmap) {
+    compressedGrayscaleImage compressed;
+    compressed.reserve(width * height);
+    for (const auto& line : bitmap) {
+        uint8_t counter = 1;
+        for (auto it = line.begin(); it != line.end(); it++) {
+            if (*it != *std::next(it) || it == line.end() - 1) {
+                compressed.push_back(std::make_pair(*it, counter));
+                counter = 1;
+            } else {
+                counter++;
+            }
+        }
+    }
+    return compressed;
+}
+grayscaleImage decompressGrayscale(const compressedGrayscaleImage& compressed) {
+    grayscaleImage decompressed;
+    size_t i = 0;
+    for (auto it : compressed) {
```

Comment:
I suggest using structured bindings introduced in C++17 for more readability: `auto [value, count] : compressed`